### PR TITLE
Add support for `fitBounds` and `fitScreenCoordinates` on globe

### PR DESCRIPTION
**Pull request by karimnaaji**
This PR adds support for `fitBounds` and `fitScreenCoordinates` with globe projection by leveraging the method introduced in #12138:
- When calling `fitBounds` or `fitScreenCoordinates`, redirect all internal calls to `_cameraForBoxAndBearing` to the newly introduced method from #12138 instead of the iterative approach performing frustum intersection by raymarching steps (which isn't adapted for globe).
- Remove the iterative method as it's now unused
- Remove warnings about unsupported API and update the docs
- Account for terrain directly in `_cameraForBounds` in order to be on par with previous behavior from the iterative frustum approach. 

Fixes https://github.com/mapbox/gl-js-team/issues/502

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [x] manually test the debug page
 - [x] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Add support for fitBounds and fitScreenCoordinates with globe projection</changelog>`

### CODE REVIEW

File: src/ui/camera.js [hunk 1]
```diff
@@ -758,10 +785,24 @@ class Camera extends Evented {
...
+
+        const z0 = this.queryTerrainElevation(coord0);
+        const z1 = this.queryTerrainElevation(coord1);
+        const z2 = this.queryTerrainElevation(coord2);
+        const z3 = this.queryTerrainElevation(coord3);
+
+        const worldCoords = [
+            [p0world.x, p0world.y, Math.min(z0 || 0, z1 || 0, z2 || 0, z3 || 0)],
+            [p1world.x, p1world.y, Math.max(z0 || 0, z1 || 0, z2 || 0, z3 || 0)]
+        ];
```
Comment (SnailBones) [inline src/ui/camera.js:804]: Looks like you're adding terrain support here, can you make a note of this in the PR?

File: src/ui/camera.js [hunk 2]
```diff
@@ -829,116 +870,16 @@ class Camera extends Evented {
         if (tr.mercatorFromTransition && zoom < halfZoomTransition) {
             tr.setProjection({name: 'globe'});
             tr.zoom = zoom;
-            return this._cameraForBoxAndBearing(tr, p0, p1, bearing, options);
+            return this._cameraForAabb(tr, p0, p1, bearing, options);
```
Comment (SnailBones) [inline src/ui/camera.js:873]: I think this name has the potential to be confused with a function taking input as 3d AABB. Maybe consider renaming it `_cameraForBounds` or `_cameraForBoundsAndBearing`?

**Review by SnailBones — APPROVED** [review]
LGTM! Just a few small nits.

File: src/ui/camera.js (hunk 1 above)
Comment (karimnaaji) [reply src/ui/camera.js:804]: Updated, this is intended to be equivalent to the previous code path and not strictly new support.

**Pull request merged**